private/pkg/transport/http/httpserver/runner.go

```diff
@@ -150,3 +149,22 @@ func httpServe(httpServer *http.Server, listener net.Listener) error {
 	}
 	return httpServer.Serve(listener)
 }
+
+// This should be the last interceptor installed.
+func newServerInterceptor() func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(writer http.ResponseWriter, request *http.Request) {
+			if len(request.Header) > 0 {
+				request = request.WithContext(
+					withIncomingHeaders(
```

Comment:
@mfridman lmk what you think here.  This is based on our discussion.  Basically keeping this server interceptor around temporarily, with a small modification. 

Now it checks to see if a header is prefixed with `rpc-` and removes it if so.  It then determines whether the remaining header key is the auth token key or the CLI version key and if so, adds them to the context.

This should hopefully handle all legacy `buf` clients since they will be passing `rpc-authorization` and `rpc-cli-version`.  Any new `buf` clients from this PR onward will be sending `authorization` and `cli-version`, which should still be caught by this interceptor.

Reply:
I am a bit confused with the underlying implementation.

This is a chi middleware that gets called very early in the request chain. There will never be `incomingHeadersContextKey{}` set on this request. The job of this middleware is to only attach a context.

I'm not suggesting we do this, but this is what I'd expect a simpler impl. to look like. Notice that we're only setting the headers with this context key. (You'll also need a getter to get at the context key since its unexported).

```go
headers := []string{
	"rpc-authorization", "authorization",
	"rpc-buf-version", "buf-version",
}
headersMap := make(map[string]string)
for _, headerName := range headers {
	if val := request.Header.Get(headerName); val != "" {
		key, val := strings.ToLower(headerName), strings.ToLower(val)
		_, after, ok := strings.Cut(key, "rpc-")
		if ok {
			key = after
		}
		if _, ok := headersMap[key]; !ok {
			headersMap[key] = val
		}
	}
}
ctx := context.WithValue(request.Context(), incomingHeadersContextKey{}, headersMap)
next.ServeHTTP(writer, request.WithContext(ctx))
```

Reply:
I still don't understand why we're attaching headers to contexts - @mfridman perhaps we should connect offline on this